client/matching/isolation_loadbalancer.go

```diff
@@ -24,124 +24,107 @@ package matching
 
 import (
 	"math/rand"
-	"slices"
-
-	"golang.org/x/exp/maps"
 
+	"github.com/uber/cadence/common/dynamicconfig"
 	"github.com/uber/cadence/common/partition"
 	"github.com/uber/cadence/common/types"
 )
 
 type isolationLoadBalancer struct {
-	provider           PartitionConfigProvider
-	fallback           LoadBalancer
-	allIsolationGroups func() []string
+	provider         PartitionConfigProvider
+	fallback         LoadBalancer
+	domainIDToName   func(string) (string, error)
+	isolationEnabled func(string) bool
 }
 
-func NewIsolationLoadBalancer(fallback LoadBalancer, provider PartitionConfigProvider, allIsolationGroups func() []string) LoadBalancer {
+func NewIsolationLoadBalancer(fallback LoadBalancer, provider PartitionConfigProvider, domainIDToName func(string) (string, error), config *dynamicconfig.Collection) LoadBalancer {
+	isolationEnabled := config.GetBoolPropertyFilteredByDomain(dynamicconfig.EnableTasklistIsolation)
 	return &isolationLoadBalancer{
-		provider:           provider,
-		fallback:           fallback,
-		allIsolationGroups: allIsolationGroups,
+		provider:         provider,
+		fallback:         fallback,
+		domainIDToName:   domainIDToName,
+		isolationEnabled: isolationEnabled,
```

Comment:
nit: reduce the footage of domainIDToName
```suggestion
		isolationEnabled: func(domainID string) bool {
            domainName, err := domainIDToName(domainID)
            if err != nil {
                return false
            }
            return isolationEnabled(domainName)
        },
```